model/src/scheduled_event/mod.rs

```diff
@@ -0,0 +1,117 @@
+//! Types for interacting with scheduled events.
+
+mod user;
+
+pub use self::user::GuildScheduledEventUser;
+
+use crate::{
+    datetime::Timestamp,
+    id::{
+        marker::{
+            ChannelMarker, GuildMarker, ScheduledEventEntityMarker, ScheduledEventMarker,
+            UserMarker,
+        },
+        Id,
+    },
+    user::User,
+};
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+/// Representation of a scheduled event.
+///
+/// For events created before October 25th, 2021, [`creator`] and [`creator_id`]
+/// will be [`None`].
+///
+/// [`creator`]: Self::creator
+/// [`creator_id`]: Self::creator_id
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct GuildScheduledEvent {
+    /// ID of the stage or voice channel if there is one.
+    ///
+    /// Present on events of [`EntityType::StageInstance`] and
+    /// [`EntityType::Voice`].
+    pub channel_id: Option<Id<ChannelMarker>>,
+    /// User object of the event's creator.
+    ///
+    /// Only present on events created after October 25th, 2021.
```

Comment:
Events can maximally be scheduled one year into the future, so it should be removed then